### PR TITLE
Add `MoNuSac` segmentation dataset

### LINKED ISSUE #546: Add MoNuSAC dataset
https://monusac-2020.grand-challenge.org/Data/

### PR DESCRIPTION
**Pull request by ioangatop**
Closes #546

image

### CODE REVIEW

File: configs/vision/dino_vit/online/monusac.yaml
Comment (nkaenzig) [inline configs/vision/dino_vit/online/monusac.yaml:1]: Please add a documentation page with the specs of the dataset.

File: configs/vision/dino_vit/online/monusac.yaml [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+          # Set `download: true` to download the dataset from https://monusac-2020.grand-challenge.org/Data/
+          # The MoNuSAC dataset is distributed under the following license: 
+          # "Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International"
+          # (see: https://creativecommons.org/licenses/by-nc-sa/4.0/legalcode)
+          transforms:
+            class_path: eva.vision.data.transforms.common.ResizeAndCrop
+            init_args:
+              mean: *NORMALIZE_MEAN
+              std: *NORMALIZE_STD
+      val:
```
Comment (nkaenzig) [inline configs/vision/dino_vit/online/monusac.yaml:93]: For datasets with only 2 splits, we often have a naming issue, because when the authors call it "test", we actually use it as a "val" dataset. Same here: https://github.com/kaiko-ai/eva/pull/547

In the leaderboard, in fact we need to report the metrics as val metrics - which is confusing because they refer to the test set. I think we need to define a clear convention here on how to solve this confusion.

Comment (nkaenzig) [inline configs/vision/dino_vit/online/monusac.yaml:93]: For this dataset, would it be possible to generate our own val split, or is it too small / data leakage issues?

**Review by nkaenzig — COMMENTED** [review]
Looks very nice. Left two minor comments.

File: configs/vision/dino_vit/online/monusac.yaml
Comment (ioangatop) [reply configs/vision/dino_vit/online/monusac.yaml:1]: good point -- on it!

File: configs/vision/dino_vit/online/monusac.yaml (hunk 1 above)
Comment (ioangatop) [reply configs/vision/dino_vit/online/monusac.yaml:98]: right yeah, I think the convention is that we use the `test` split as validation, so in leaderboard we will report it as `val` -- we treat the MHIST dataset the same

Comment (ioangatop) [reply configs/vision/dino_vit/online/monusac.yaml:98]: we can avoid the data leakage, as they do provide the patient id, however the dataset is indeed small; train 209 sample and test 85

File: configs/vision/dino_vit/online/monusac.yaml
Comment (ioangatop) [reply configs/vision/dino_vit/online/monusac.yaml:1]: done!

File: docs/datasets/monusac.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+The dataset class `MoNuSAC` supports downloading the data during runtime by setting the init argument `download=True`.
+
+> [!NOTE]
+> In the provided `MoNuSAC`-config files the download argument is set to `false`. To enable automatic download you will need to open the config and set `download: true`.
+
+### Splits
+
+We work with the splits provided by the data source.
+
+| Splits   | Train           | Test         |
```
Comment (roman807) [inline docs/datasets/monusac.md:51]: if we don't have a validation set and validate on the test set, lets refer to the test set as validation set (as we did e.g. for mhist: https://kaiko-ai.github.io/eva/latest/datasets/mhist/)

Comment (roman807) [reply docs/datasets/monusac.md:51]: also best to align the language (and remove note in line 56/57), since were describing the same issue

File: configs/vision/dino_vit/online/monusac.yaml (hunk 1 above)
Comment (roman807) [reply configs/vision/dino_vit/online/monusac.yaml:108]: since some images exceed 224x224, would it make sense to do patching (like here: https://github.com/kaiko-ai/eva/pull/547/files#diff-d0df374ae39abf8b6d321c9334cfcfcd8d41aa0e344daa741fdb6df95f43f5b7) which would increase the dataset / potentially allow for a val split?

File: docs/datasets/monusac.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+│   │   ├── TCGA-5P-A9K0-01Z-00-DX1_1.tif   # tissue image
+│   │   ├── TCGA-5P-A9K0-01Z-00-DX1_1.xml   # annotations
+│   │   └── ...
+```
+
+## Download and preprocessing
+The dataset class `MoNuSAC` supports downloading the data during runtime by setting the init argument `download=True`.
+
+> [!NOTE]
+> In the provided `MoNuSAC`-config files the download argument is set to `false`. To enable automatic download you will need to open the config and set `download: true`.
```
Comment (roman807) [inline docs/datasets/monusac.md:45]: Lets align the Note style and language with those of the other datasets (if you prefer this one, change the others as well)

File: docs/datasets/monusac.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
+# MoNuSAC
```
Comment (roman807) [inline docs/datasets/monusac.md:1]: add this file to `mkdocs.yaml`

File: src/eva/vision/data/datasets/segmentation/monusac.py [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+                )
+                semantic_labels[fill_col_coords, fill_row_coords] = class_id
+
+        return semantic_labels
+
+    def _download_dataset(self) -> None:
+        """Downloads the dataset."""
+        self._print_license()
+        for resource in self._resources:
+            if os.path.isdir(self._root):
```
Comment (roman807) [inline src/eva/vision/data/datasets/segmentation/monusac.py:227]: looks like this will only download the first dataset (train), and then skip the test because then `self._root` exists

Comment (ioangatop) [reply src/eva/vision/data/datasets/segmentation/monusac.py:227]: ah good point -- fixed!

File: configs/vision/dino_vit/online/monusac.yaml (hunk 1 above)
Comment (ioangatop) [reply configs/vision/dino_vit/online/monusac.yaml:109]: Not sure... now I do random crops so it should capture all the regions, so not sure we need it;  maybe on the offline if we dont go `CenterCrop` transform -- we can investigate in the future though

File: docs/datasets/monusac.md (hunk 2 above)
Comment (ioangatop) [reply docs/datasets/monusac.md:51]: ok done

File: docs/datasets/monusac.md (hunk 3 above)
Comment (ioangatop) [reply docs/datasets/monusac.md:45]: this is the official markdown note style, so probably nice to follow -- ok I update all of them

File: docs/datasets/monusac.md (hunk 4 above)
Comment (ioangatop) [reply docs/datasets/monusac.md:1]: nice done

**Review by roman807 — APPROVED** [review]
